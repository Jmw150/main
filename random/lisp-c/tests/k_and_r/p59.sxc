(#include <stdio.h>)

// count digits, white space, others
(int main ()
     (var int c i nwhite nother ([] ndigit 10))

     (= nwhite nother 0)

     (for (= i 0) (< i 10) (i ++)
	  (= ([] ndigit i) 0))

     (while (!= (= c (getchar)) EOF)
       (switch c
	       (('0 '1 '2 '3 '4 '5 '6 '7 '8 '9)
		(([] ndigit (- c '0)) ++)
		break)
	       (('  '\n '\t)
		(nwhite ++)
		break)
	       (default
		   (nother ++)
		   break)))
     (printf "digits =")
     (for (= i 0) (< i 10) (i ++)
	  (printf " %d" ([] ndigit i)))
     (printf ", white space = %d, other = %d\n" nwhite nother)
     (return 0))
