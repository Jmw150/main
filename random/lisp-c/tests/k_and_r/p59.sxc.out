digits = 9 3 0 0 0 0 0 0 0 1, white space = 159, other = 345
